tests/components/normal-components/chip-pin-alias.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { expect, it } from "bun:test"
+import "lib/register-catalogue"
+import { getTestFixture } from "tests/fixtures/get-test-fixture"
+
+it("Chip with pin labels as strings and duplicates", async () => {
+  const { circuit } = getTestFixture()
+
+  circuit.add(
+    <board width="10mm" height="10mm">
+      <chip
+        name="U1"
+        schPinSpacing={0.75}
+        schPortArrangement={{
+          leftSide: {
+            direction: "top-to-bottom",
+            pins: ["B2", "B1"],
+          },
+          rightSide: {
+            direction: "top-to-bottom",
+            pins: ["A1", "A2"],
+          },
+          topSide: {
+            direction: "left-to-right",
+            pins: ["B1", "B2"],
+          },
+          bottomSide: {
+            direction: "left-to-right",
+            pins: ["B3", "B3"],
+          },
+        }}
+        supplierPartNumbers={{
+          lcsc: ["C165948"],
+        }}
+      />
+    </board>,
+  )
+
+  circuit.render()
+
+  const fs = require("fs")
+  fs.writeFileSync("circuit.json", JSON.stringify(circuit.getCircuitJson(), null, 2))
+  expect(circuit.getCircuitJson()).toMatchSchematicSnapshot(import.meta.path)
```

Comment:
👏 👏 👏 👏  this is a really good sample. We should remember to remove the file writing but looks good (ikik this is a draft) but looks good 👍

Reply:
this test should _not_ rely on snapshots however!!!! Load in all the `source_port`s, sort them, and confirm in code that they match expectations. 

FWIW i think this test in particular points to a very confusing behavior from the normalizer. `B11` and `B112` should probably be `B1_1` and `B1_2`!!! We can fix it in a future PR and look for these edge cases with the normalizer- but really this test reproduces a bug and we should leave a comment in the test that indicates that it's a known issue that should be fixed